nb2kg/handlers.py

```diff
@@ -86,8 +89,23 @@ def get(self, kernel_id, *args, **kwargs):
         self.kernel_id = cast_unicode(kernel_id, 'ascii')
         yield gen.maybe_future(super(WebSocketChannelsHandler, self).get(kernel_id=kernel_id, *args, **kwargs))
 
+    def send_ping(self):
+        if self.stream.closed() and self.ping_callback is not None:
+            self.ping_callback.stop()
+            return
+
+        self.ping(b'')
+
+    def on_pong(self, data):
+        self.log.debug('Client has sent pong')
+
     def open(self, kernel_id, *args, **kwargs):
         """Handle web socket connection open to notebook server and delegate to gateway web socket handler """
+        loop = IOLoop.current()
+
+        self.ping_callback = PeriodicCallback(self.send_ping, WS_PING_INTERVAL_MS, io_loop=loop)
```

Comment:
Please remove the `io_loop` parameter.  It was removed in Tornado 5.0 and deprecated in 4.1.  It currently produces the following repeatedly on systems running Tornado >= 5.0...
```
[E 07:53:09.523 NotebookApp] Uncaught exception GET /api/kernels/e26308d6-517d-40b3-a5ef-3da204c73923/channels?session_id=70fb168f96ef449a80de44728d1f906d (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/api/kernels/e26308d6-517d-40b3-a5ef-3da204c73923/channels?session_id=70fb168f96ef449a80de44728d1f906d', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/opt/anaconda3/envs/nb2kg-dev/lib/python3.7/site-packages/tornado/websocket.py", line 952, in _accept_connection
        open_result = handler.open(*handler.open_args, **handler.open_kwargs)
      File "/opt/anaconda3/envs/nb2kg-dev/lib/python3.7/site-packages/nb2kg/handlers.py", line 106, in open
        self.ping_callback = PeriodicCallback(self.send_ping, WS_PING_INTERVAL_MS, io_loop=loop)
    TypeError: __init__() got an unexpected keyword argument 'io_loop'
```
Once removed, I see expected behavior for startup (although I didn't test behind a proxy).